### PR TITLE
verify exists index 0 in $ref

### PR DESCRIPTION
**Pull request by phlyper**
I have to get the referrer url from any page if exists

exemple

```
class AppController extends Controller {
......
    public function beforeRender() {
        $this->params['referer'] = $this->referer(null, true);
    }
..........
}
```

### CODE REVIEW

**Comment by markstory** [comment]
When does '0' not exist?

**Comment by dereuromark** [comment]
when
`$ref = ''`
would be my guess.

**Comment by ADmad** [comment]
@phlyper Please provide an example to reproduce the error.

**Comment by phlyper** [comment]
@ADmad screenshot for the error

cakephp screen

File: lib/Cake/Network/CakeRequest.php [hunk 1]
```diff
@@ -448,7 +448,7 @@ public function referer($local = false) {
 		if (!empty($ref) && !empty($base)) {
 			if ($local && strpos($ref, $base) === 0) {
 				$ref = substr($ref, strlen($base));
-				if ($ref[0] !== '/') {
+				if (isset($ref[0]) && $ref[0] !== '/') {
```
Comment (dereuromark) [inline lib/Cake/Network/CakeRequest.php]: I would simply use `!empty($ref) && ...`.

Comment (markstory) [inline lib/Cake/Network/CakeRequest.php]: I'll fix that up post merge.

**Pull request merged**